Storage/tests/System/KmsTest.php

```diff
@@ -0,0 +1,279 @@
+<?php
+/**
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Storage\Tests\System;
+
+use Google\Cloud\Core\Exception\ConflictException;
+use Google\Cloud\Core\RequestWrapper;
+use Google\Cloud\Storage\StorageObject;
+use GuzzleHttp\Psr7\Request;
+
+/**
+ * @group storage
+ * @group storage-kms
+ */
+class KmsTest extends StorageTestCase
+{
+    const DATA = 'data';
+    const KEY_RING_ID = 'kms-kr';
+    const CRYPTO_KEY_ID_1 = 'key1';
+    const CRYPTO_KEY_ID_2 = 'key2';
+
+    private static $keyName1;
+    private static $keyName2;
+
+    public static function setUpBeforeClass()
+    {
+        parent::setUpBeforeClass();
+        list(self::$keyName1, self::$keyName2) = self::getKeyNames(
+            self::KEY_RING_ID,
+            self::CRYPTO_KEY_ID_1,
+            self::CRYPTO_KEY_ID_2
+        );
+    }
+
+    public function testUpload()
+    {
+        $object = $this->upload();
+
+        $this->assertContains(self::$keyName1, $object->info()['kmsKeyName']);
+        $this->assertEquals(self::DATA, $object->downloadAsString());
+    }
+
+    public function testUploadWithDefaultNameOnBucket()
+    {
+        self::$bucket->update([
+            'encryption' => [
+                'defaultKmsKeyName' => self::$keyName1
+            ]
+        ]);
+
+        $object = $this->upload(['metadata' => null]);
+
+        $this->assertContains(self::$keyName1, $object->info()['kmsKeyName']);
+        $this->assertEquals(self::DATA, $object->downloadAsString());
+
+        // Reset default to none
+        self::$bucket->update([
```

Comment:
Any benefit in verifying that this removes the encryption settings as expected? (i.e. another bucket reload and inspect response?)